### PR TITLE
Indexes

### PR DESCRIPTION
**Pull request by ncarazon**
- setup hardcoded data for index scores
- implemented index view for notebook page when matching index slug
- implemented basic table component, implemented indexes table with column sorting (desktop, mobile views)
- implemented index score calculation formula
- implemented weekly movement component, integrated it to index table and feed card

image

image

### CODE REVIEW

**Review by elisescu — APPROVED** [review]
Nice, LGTM!

File: front_end/src/app/(main)/notebooks/components/index_notebook/styles.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[data-editor-block-type="image"] img {
+  @apply !max-h-none;
+}
```
Comment (hlbmtc) [inline front_end/src/app/(main)/notebooks/components/index_notebook/styles.css:3]: EOF

**Review by hlbmtc — APPROVED** [review]

**Comment by hickman-santini** [comment]
@ncarazon a few requests:

- [ ] fwiw I get bad gateway when I clicked on [this question](https://dev.metaculus.com/questions/27881/chinas-youth-unemployment-rate-for-aug-2024/), I assume that's a dev environment weirdness
- [ ] For continuous questions, CP change should be actual change in the CP median e.g. if it went from 4.5 to 3.3, it should say :arrow_down: 1.2 this week (this might already be the case; can you just show an example?)
- [ ] For binaries... I want it to be clear this is **change in percentage points**, not percent change. I think we should spell out "percentage points" instead of "%" even though it will be long
- [ ] This wasn't in the design, but can you add change this week to the index value? Same format as for continuous questions (just bigger font... but not as big as the index value text)

**Comment by aseckin** [comment]
image
Here's the suggested change for the Index Value part - imo it helps when it looks like a part of the table. 

For the movement in Index Value, are we using absolute values?

**Comment by ncarazon** [comment]
> For the movement in Index Value, are we using absolute values?

No, we're not

**Comment by ncarazon** [comment]
> fwiw I get bad gateway when I clicked on [this question](https://dev.metaculus.com/questions/27881/chinas-youth-unemployment-rate-for-aug-2024/), I assume that's a dev environment weirdness

Yeah, not related to Indexes itself. Maybe @hlbmtc would have more context on what could be the reason?

> For continuous questions, CP change should be actual change in the CP median e.g. if it went from 4.5 to 3.3, it should say ⬇️ 1.2 this week (this might already be the case; can you just show an example?)

This is the way it works, though, I pushed a minor fix. Here is an example:

image

image

image

**Comment by hickman-santini** [comment]
Remove "%" from "X% percentage points" - otherwise LGTM! The questions aren't in yet. When they are, I'll ask you to sub that series in and then we'll push to prod? 

On a related note:

There will be a news piece [like this](https://www.metaculus.com/notebooks/30299/threshold-2030-modeling-ai-economic-futures/) about the Curve workshop that generated this index, which will also generally introduce the index feature as something we're experimenting with. I would like to be able to embed the index table from this project page in that news page. Would that be straightforward or no? Is there anything that constrains news pages to being static?

**Comment by ncarazon** [comment]
> The questions aren't in yet. When they are, I'll ask you to sub that series in and then we'll push to prod?

I'll merge the PR now, as Indexes presentation is only applied to `agi-readiness-index` notebook. When questions are ready we'll create this notebook on prod and I'll update question ids and weights configuration.

> There will be a news piece [like this](https://www.metaculus.com/notebooks/30299/threshold-2030-modeling-ai-economic-futures/) about the Curve workshop that generated this index, which will also generally introduce the index feature as something we're experimenting with. I would like to be able to embed the index table from this project page in that news page. Would that be straightforward or no? Is there anything that constrains news pages to being static?

This won’t be straightforward since the indexes table isn’t part of the markdown content. However, it is possible to add support for embedding the table into markdown, though it will require some effort. Let me know if you’d like me to implement this, and I can handle it in a separate PR.

An alternative approach could be embedding a table screenshot, similar to what we’re planning for the Index graph on the notebook page. However, this wouldn’t be dynamic, so it might not fully meet our needs.

**Pull request merged**